lib/screens/home/products/widgets/category_filter_dropdown_menu.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import 'package:shop_app/models/product/models.dart';
+import 'package:shop_app/screens/home/products/products_bloc.dart';
+
+class CategoryFilterDropDownMenu extends StatelessWidget {
+  const CategoryFilterDropDownMenu({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<ProductsBloc, ProductsState>(
```

Comment:
Use BlocSelector<ProductsBloc, ProductsState, Category?> to optimize UI rebuilds